plugins/org.jboss.tools.livereload.core/src/org/jboss/tools/livereload/core/internal/util/WSTUtils.java

```diff
@@ -571,5 +568,39 @@ private static String getContextRoot(final IServer server, final IModule module)
 	private static IURLProvider getURLProvider(final IServer server) {
 		return (IURLProvider) server.loadAdapter(IURLProvider.class, new NullProgressMonitor());
 	}
+	
+	/**
+	 * Gets the {@link ServerExtendedProperties} for the given {@code server}. 
+	 * @param server the server to analyze
+	 * @return the associated {@link ServerExtendedProperties} or <code>null</code> if none exists.
+	 */
+	private static ServerExtendedProperties getServerExtendedProperties(final IServer server) {
+		return (ServerExtendedProperties) server.loadAdapter(ServerExtendedProperties.class, new NullProgressMonitor());
+	}
+	
+	/**
+	 * Gets the {@link URL} of the deployed {@code module}  deployed on the given {@code server}.
+	 * @param server the {@link IServer} on which the module is deployed
+	 * @param module the {@link IModule} whose context root is to be found
+	 * @return the URL of the {@code module} on the {@code server} or
+	 * 		   {@code null} if it is not a web module
+	 * 					    or if this server does not have a valid configuration
+	 */
+	public static URL getModuleURL(IServer server, IModule module) {
+		ServerExtendedProperties props = getServerExtendedProperties(server);
+		if (props == null || isEjbModule(module)) {
+			return null;
+		}
+		final IURLProvider urlProvider = getURLProvider(server);
+		if (urlProvider != null) {
+			return urlProvider.getModuleRootURL(module);
+		}
+		return null;
+	}
+	
+	private static boolean isEjbModule(IModule module) {
```

Comment:
@xcoulon maybe. I think we need some more opinions. Anyway we need to make it everywhere the same way (see links of [the comment](https://github.com/jbosstools/jbosstools-livereload/pull/129#issuecomment-227119501) above)